<launch>
  <!--
      This is a launch file for Main PC of Field Computer (FC)
  -->

  <arg name="USE_COM" default="true" />
  <node pkg="drc_task_common" type="fc-executive.l" name="fc_executive" output="screen">
  </node>
  <node pkg="drc_task_common" type="robot-main.l" name="robot_main" output="screen" if="false">
  </node>

  <!-- recognition -->
  <group ns="stereo_preprocessing">
    <include file="$(find drc_task_common)/launch/fc/stereo_preprocess.launch">
      <arg name="INPUT_POINT_CLOUD" value="/multisense/resize_1_4/points" />
    </include>
  </group>
  <group ns="go_pos_recognition">
    <include file="$(find drc_task_common)/launch/fc/centroid_of_pointcloud_in_rect.launch">
    </include>
  </group>
  <group ns="look_at_recognition">
    <include file="$(find drc_task_common)/launch/fc/centroid_of_pointcloud_in_rect.launch">
    </include>
  </group>
  <include file="$(find drc_task_common)/launch/fc/valve_recognition.launch">
  </include>
  <include file="$(find drc_task_common)/launch/fc/door_recognition.launch">
    <arg name="INPUT_HINT_POINT_CLOUD" value="/stereo_preprocessing/normal_estimation/output_with_xyz" />
    <arg name="INPUT_POINT_CLOUD" value="/stereo_preprocessing/normal_estimation/output_with_xyz" />
  </include>
  <!-- communication -->
  <include if="$(arg USE_COM)"
           file="$(find drc_com_common)/launch/field_computer_com.launch" />
  <group if="false">

  <!-- object detection -->
  <include file="$(find drc_task_common)/launch/object_detect.launch" />

  <!-- remote server -->
  <include file="$(find drc_task_common)/launch/remote_server.launch" />
  </group>
</launch>
